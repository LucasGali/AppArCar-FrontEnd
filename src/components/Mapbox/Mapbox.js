import React, { useRef, useState, useCallback } from 'react';
import ReactMapGL, { GeolocateControl, NavigationControl, Marker } from 'react-map-gl';
import './Mapbox.css';
import location from '../../img/location.png';
import { connect } from 'react-redux';

const geolocateControlStyle = {
  right: 10,
  top: 10,
};

const navControlStyle = {
  right: 10,
  top: 45,
};

const Mapbox = ({ parkingLost }) => {
  const [viewport, setViewport] = useState({
    width: '101%',
    height: '100%',
    latitude: -26.8307,
    longitude: -65.2036,
    zoom: 14,
  });
  const mapref = useRef(viewport);
  const handleViewportChange = useCallback(
    (newViewport) => setViewport(newViewport),
    [],
    );
    
  const MarkerComponent = () => {
    return parkingLost.parkings.map(lot => 
      <Marker latitude={lot.latitude} longitude={lot.longitude} key={lot._id} >
        <img alt='location' className='marker-btn' src={location} />
      </Marker>       
    )
  };

  return (
    <ReactMapGL
      mapStyle={'mapbox://styles/mapbox/streets-v11'}
      mapboxApiAccessToken={process.env.REACT_APP_MAPBOX_TOKEN}
      {...viewport}
      ref={mapref}
      onViewportChange={handleViewportChange}
    >
      <GeolocateControl
        style={geolocateControlStyle}
        positionOptions={{ enableHighAccuracy: true }}
        trackUserLocation={true}
        auto
      />
      <NavigationControl style={navControlStyle} />
      <MarkerComponent />
    </ReactMapGL>
  );
};

const mapStateToProps = (state) => ({
  parkingLost: state.parkingData,
});

export default connect(mapStateToProps)(Mapbox);
