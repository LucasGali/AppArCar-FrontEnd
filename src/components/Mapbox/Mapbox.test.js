import { render } from '@testing-library/react';
import Mapbox from './Mapbox';


describe('Mapbox Component', () => {
  it('renders default form acording to snapshot', () => {
    const { asFragment } = render(<Mapbox />);
    expect(asFragment(<Mapbox />)).toMatchSnapshot();
  });
});
