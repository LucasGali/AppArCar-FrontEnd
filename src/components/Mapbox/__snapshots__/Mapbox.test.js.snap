// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mapbox Component renders default form acording to snapshot 1`] = `
<DocumentFragment>
  <div
    style="position: relative; width: 101%; height: 100%; cursor: grab; outline: none;"
    tabindex="0"
  >
    <div
      style="position: relative; width: 100%; height: 100%;"
    >
      <div
        class=""
        style="position: absolute; width: 100%; height: 100%; overflow: hidden; visibility: inherit;"
      />
    </div>
  </div>
</DocumentFragment>
`;
